frontend/components/transaction/SwapTransaction.vue

```diff
@@ -29,27 +29,39 @@
             </Explain>
             the auction. In case your transaction will be rejected, it only results in the loss of the transaction fee.
         </TextBlock>
-        <WalletBlock
-            class="mb-6 lg:mb-0"
-            :disabled="!auctionTransaction.isActive"
-            :is-loading="isConnecting"
+        <WalletConnectionCheckPanel
             :wallet-address="walletAddress"
+            :is-loading="isConnecting"
             :is-explanations-shown="isExplanationsShown"
+            :disabled="!auctionTransaction.isActive"
             @connectWallet="$emit('connect')"
             @disconnectWallet="$emit('disconnect')"
         />
-        <AuthorisationBlock
-            class="mb-6 lg:mb-0"
-            :disabled="!isConnected || !auctionTransaction.isActive"
-            :auction-transaction="auctionTransaction"
-            :is-loading="isAuthorizing"
+        <WalletAuthorizationCheckPanel
             :is-wallet-authorized="isWalletAuthorized"
-            :is-collateral-authorised="isCollateralAuthorised"
+            :disabled="!auctionTransaction.isActive"
+            :wallet-address="walletAddress"
             :is-explanations-shown="isExplanationsShown"
+            :is-loading="isAuthorizing"
             @authorizeWallet="$emit('authorizeWallet')"
+        />
+        <CollateralAuthorizationCheckPanel
+            :collateral-type="auctionTransaction.collateralType"
+            :is-collateral-authorized="isCollateralAuthorised"
+            :auth-transaction-fee-e-t-h="auctionTransaction.authTransactionFeeETH"
+            :wallet-address="walletAddress"
+            :is-loading="isAuthorizing"
+            :disabled="!auctionTransaction.isActive || !isWalletAuthorized"
+            :is-explanations-shown="isExplanationsShown"
             @authorizeCollateral="$emit('authorizeCollateral', auctionTransaction.collateralType)"
         />
+        <GrossProfitCheckPanel :gross-profit="auctionTransaction.transactionGrossProfit" />
+        <NetProfitCheckPanel
+            :net-profit="auctionTransaction.transactionNetProfit"
+            :negative-gross-profit="!auctionTransaction.transactionGrossProfit.isGreaterThanOrEqualTo(0)"
+        />
         <ExecutionBlock
```

Comment:
Please pass separate `isCorrect` values into the `disabled` prop of the `ExecutionBlock` (you also would need to clean up `ExecutionBlock` to not accept any extra props it doesn't need, eg `transaction-gross-profit`)

Reply:
I removed `transaction-gross-profit`, but `wallet-address` is needed in enable the send profit to another wallet feature.